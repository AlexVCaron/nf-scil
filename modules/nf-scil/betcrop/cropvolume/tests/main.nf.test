nextflow_process {

    name "Test Process BETCROP_CROPVOLUME"
    script "../main.nf"
    process "BETCROP_CROPVOLUME"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "betcrop"
    tag "betcrop/cropvolume"

    test("betcrop - cropvolume") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['betcrop']['cropvolume']['image'], checkIfExists: true),
                    []
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("betcrop - cropvolume - outputbbox") {

        config "./nextflow_bbox.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['betcrop']['cropvolume']['image'], checkIfExists: true),
                    []
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
