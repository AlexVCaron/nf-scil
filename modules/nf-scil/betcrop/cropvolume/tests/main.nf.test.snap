{
    "betcrop - cropvolume - outputbbox": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,9d98c302556a95d0bec8f26be9d2f9cd"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,47575e3531416000b55551ea9b562ce2"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7eb8e64496eab011bfb2b89f9035729a"
                ],
                "bounding_box": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,47575e3531416000b55551ea9b562ce2"
                    ]
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,9d98c302556a95d0bec8f26be9d2f9cd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7eb8e64496eab011bfb2b89f9035729a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T16:35:04.627902"
    },
    "betcrop - cropvolume": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,9d98c302556a95d0bec8f26be9d2f9cd"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,7eb8e64496eab011bfb2b89f9035729a"
                ],
                "bounding_box": [
                    
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,9d98c302556a95d0bec8f26be9d2f9cd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7eb8e64496eab011bfb2b89f9035729a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T16:34:52.36467"
    }
}